resources/assets/lib/components/report-form.tsx

```diff
@@ -0,0 +1,153 @@
+    }
+
+    this.state = {
+      comments: '',
+      selectedReason: this.options[0],
+    };
+  }
+
+  render() {
+    return this.props.visible ? this.renderForm() : null;
+  }
+
+  private handleCommentsChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    this.setState({ comments: e.target.value });
+  };
+
+  private handleReasonChange = (option: ReportOption) => {
+    this.setState({ selectedReason: option });
+  };
+
+  private renderForm() {
+    const title = this.props.completed ? osu.trans('users.report.thanks') : this.props.title;
+
+    return (
+      <Modal onClose={this.props.onClose} visible={this.props.visible}>
+        <div className={bn}>
+          <div className={`${bn}__header`}>
+            <div className={`${bn}__row ${bn}__row--exclamation`}>
+              <i className='fas fa-exclamation-triangle' />
+            </div>
+
+            <div className={`${bn}__row ${bn}__row--title`} dangerouslySetInnerHTML={{ __html: `<span>${title}</span>`}} />
+          </div>
+
+          {this.renderFormContent()}
+        </div>
+      </Modal>
+    );
+  }
+
+  private renderFormContent() {
+    const reasonElements = isEmpty(this.options) ? null :
+      (
+        <>
+          <div className={`${bn}__row`}>
+            {osu.trans('users.report.reason')}
+          </div>
+          <div className={`${bn}__row`}>
+            <SelectOptions
+              blackout={false}
+              bn={`${bn}-select-options`}
+              onChange={this.handleReasonChange}
+              options={this.options}
+              selected={this.state.selectedReason}
+            />
+          </div>
+        </>
+      );
+
+    return (
+      <div>
+        {reasonElements}
+        <div className={`${bn}__row`}>
+          {osu.trans('users.report.comments')}
+        </div>
+        <div className={`${bn}__row`}>
+          <textarea
+            className={`${bn}__textarea`}
+            maxLength={maxLength}
+            onChange={this.handleCommentsChange}
+            placeholder={osu.trans('users.report.placeholder')}
+            value={this.state.comments}
+          />
+        </div>
+        <div className={`${bn}__row ${bn}__row--buttons`}>
+          <button
+            className={`${bn}__button ${bn}__button--report`}
+            disabled={this.props.disabled || this.state.comments.length === 0}
+            onClick={this.sendReport}
+            type='button'
+          >
+            {osu.trans('users.report.actions.send')}
+          </button>
+          <button
+            className={`${bn}__button`}
+            disabled={this.props.disabled}
+            onClick={this.props.onClose}
+            type='button'
+          >
+            {osu.trans('users.report.actions.cancel')}
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  private sendReport = (_e: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
Should be declared `readonly`.

Params can be omitted if none of them are used.